reggie/ingestion/preprocessor/west_virginia_preprocessor.py

```diff
@@ -1,87 +1,361 @@
+        df_voters.insert(
+            1,
+            "County_ID",
+            df_voters.County_Name.map(slugify),
         )
 
-        # --- handling voter file --- #
+        # Make sure counties are valid as defined in config
+        # counties = df_voters.County_ID.unique()
+        self.check_column_has_valid_values(
+            df_voters,
+            "County_ID",
+            self.config_lookup_to_valid_list(config["locales_counties"]),
+        )
 
-        gender_conversion_dict = {}
-        for c, v in self.config["gender_codes"].items():
-            for i in v:
-                if i is None:
-                    gender_conversion_dict[" "] = c
-                gender_conversion_dict[i] = c
+        # Gender clean
+        gender_dict = self.config_lookup_to_dict(self.config["gender_codes"])
+        df_voters.loc[:, "SEX"] = df_voters.loc[:, "SEX"].map(gender_dict)
 
-        df_voter.loc[:, "SEX"] = df_voter.loc[:, "SEX"].map(
-            gender_conversion_dict
+        # Gender check
+        self.check_column_has_valid_values(
+            df_voters, "SEX", self.config_lookup_to_valid_list(config["gender_codes"])
         )
 
-        df_voter = self.config.coerce_dates(df_voter)
-        df_voter = self.config.coerce_strings(
-            df_voter, exclude=[self.config["voter_id"]]
+        # Party clean
+        df_voters.PartyAffiliation = df_voters.PartyAffiliation.apply(
+            self.clean_party_value
         )
 
-        # coerce_strings does not convert party_identifier but conversion is needed in this instance
-        df_voter.loc[:, self.config["party_identifier"]] = (
-            df_voter.loc[:, self.config["party_identifier"]]
-            .str.replace("\W", " ")
-            .str.strip()
-        )
+        # Status clean up
+        status_dict = self.config_lookup_to_dict(self.config["status_codes"])
+        df_voters.loc[:, "Status"] = df_voters.loc[:, "Status"].map(status_dict)
 
-        df_voter = df_voter.set_index(self.config["voter_id"])
+        # Clean up districts by removing leading zeroes, but keep as strings
+        for field in [
+            "Congressional District",
+            "Senatorial District",
+            "Delegate District",
+            "Magisterial District",
+            "Precinct_Number",
+        ]:
+            df_voters[field] = df_voters[field].apply(
+                lambda x: x.lstrip("0") if isinstance(x, str) else x
+            )
 
-        self.meta = {
-            "message": "west_virginia_{}".format(datetime.now().isoformat())
-            # vote history not available
-            #            'array_encoding': json.dumps(),
-            #            'array_decoding': json.dumps()
-        }
+        # Coerce dates
+        df_voters = self.config.coerce_dates(df_voters)
+
+        # Set index
+        # TODO: Is this necessary?  Is this helpful higher up?
+        df_voters = df_voters.set_index(config["voter_id"], drop=False)
```

Comment:
VS doesn't need the voter_id to be in any particular place in the processed file, although it is nice if it's the first column. I think Delaney usually does this to make her history merging easier, but you don't have to do it. Just note that your decision here about dropping the voter_id or not impacts whether you actually want to write out the index or not in the data write below on line 247:

io_obj=StringIO(df_voters.to_csv(encoding="utf-8", **index=True**)),